flask_server.py

```diff
@@ -110,6 +84,57 @@ def logs():
 
 ################### API ROUTES ###################
 
+def get_index_settings(user_settings):
+        if scraper_runner.args:
+            loop_mode = not scraper_runner.shutting_down
+            media_mode = scraper_runner.args['media_mode']
+            ids_source = scraper_runner.args['ids_source']
+        else:
+            loop_mode = False
+            media_mode = "all"
+            ids_source = "all"
+
+        status = {
+            "log_lines": get_log_file_list(),
+            "logged_in": True if "session_id" in user_settings else False,
+            "output": scraper_runner.getOutput(),
+            "scraper_settings": {
+                "loop_mode": loop_mode,
+                "media_mode": media_mode,
+                "ids_source": ids_source
+            },
+            "scraper_status": scraper_runner.getStatus()
+        }
+
+        return jsonify(status)
+
+@app.route("/api/index/", methods=['POST', 'GET'])
+def index_api():
+    logger.info(f"API/{request.method} request to /index/")
+    user_settings = settings.get()
+    if request.method == 'GET':  # Returns the state of the scraper
+        return get_index_settings(user_settings)
+
+    elif request.method == 'POST':  # Starts/stops the scraper
+        res = request.get_json()
+        if not "session_id" in user_settings:
+            return {"status": "not logged in"}
+
+        if res['command'] == "start":
+            scraper_settings = deepcopy(res)
+            scraper_settings.pop('loop_mode')
+            scraper_settings.pop('command')
```

Comment:
Please don't, find a different way. If you separate the api it should be easier anyway